### PR TITLE
dev-python/entrypoints: Version bump to 0.3 + dependency fixes

### PR DESCRIPTION
**Pull request by marbre**
Package-Manager: Portage-2.3.66, Repoman-2.3.11

### CODE REVIEW

**Comment by marbre** [comment]
I would really appreciate if this request gets merged as this version bump also fixes two reported bugs.

File: dev-python/entrypoints/files/entrypoints-0.3-setup.py.patch [hunk 1]
```diff
@@ -0,0 +1,20 @@
+setup.py for entrypoints.
```
Comment (a17r) [inline dev-python/entrypoints/files/entrypoints-0.3-setup.py.patch]: Why is this not being supplied by upstream though?

File: dev-python/entrypoints/entrypoints-0.3.ebuild [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+python_prepare_all() {
+	# Prevent un-needed download during build
+	if use doc; then
+		sed -e "/^    'sphinx.ext.intersphinx',/d" -i doc/conf.py || die
+	fi
+
+	distutils-r1_python_prepare_all
+
+	mv "${WORKDIR}/${P}"/entrypoints.py "${WORKDIR}/${P}/${PN}/" || die
```
Comment (a17r) [inline dev-python/entrypoints/entrypoints-0.3.ebuild]: why is this line necessary?

File: dev-python/entrypoints/entrypoints-0.3.ebuild [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	$(python_gen_cond_dep '>=dev-python/configparser-3.5.0[${PYTHON_USEDEP}]' 'python2*')
+	test? (
+		dev-python/pytest[${PYTHON_USEDEP}]
+		virtual/python-pathlib[${PYTHON_USEDEP}]
+	)
+	doc? ( dev-python/sphinx[${PYTHON_USEDEP}] )
+	"
+
+PATCHES=(
+	"${FILESDIR}/${P}"-setup.py.patch
```
Comment (a17r) [inline dev-python/entrypoints/entrypoints-0.3.ebuild]: this could also be distributed as file rather than patch

File: dev-python/entrypoints/entrypoints-0.3.ebuild [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+KEYWORDS="~amd64 ~arm ~arm64 ~x86"
+IUSE="doc test"
+
+DEPEND="
+	$(python_gen_cond_dep '>=dev-python/configparser-3.5.0[${PYTHON_USEDEP}]' 'python2*')
+	test? (
+		dev-python/pytest[${PYTHON_USEDEP}]
+		virtual/python-pathlib[${PYTHON_USEDEP}]
+	)
+	doc? ( dev-python/sphinx[${PYTHON_USEDEP}] )
```
Comment (a17r) [inline dev-python/entrypoints/entrypoints-0.3.ebuild]: dependencies ought to be sorted, so `doc` should come before `test`

File: dev-python/entrypoints/files/entrypoints-0.3-setup.py.patch (hunk 1 above)
Comment (marbre) [reply dev-python/entrypoints/files/entrypoints-0.3-setup.py.patch]: The upstream repository just does not provide a setup.py. Anyway, we could switch over and download the archive from pypi instead of github, as the archive there contains an setup.py generated by flit. Cannot remember if there was anything against this before.

File: dev-python/entrypoints/entrypoints-0.3.ebuild (hunk 2 above)
Comment (marbre) [reply dev-python/entrypoints/entrypoints-0.3.ebuild]: Without this line only `__init__.py` is installed to `site-packages/entrypoints` and `entrypoints.py` is not.

File: dev-python/entrypoints/entrypoints-0.3.ebuild (hunk 3 above)
Comment (marbre) [reply dev-python/entrypoints/entrypoints-0.3.ebuild]: Yeah, don't know it was decided/accepted more than two years ago to go with a patch. Might be obsolete anyway.

File: dev-python/entrypoints/entrypoints-0.3.ebuild (hunk 4 above)
Comment (marbre) [reply dev-python/entrypoints/entrypoints-0.3.ebuild]: Will do so within the updated PR, pushing as soon as I know the way to go with the setup.py issue.

File: dev-python/entrypoints/files/entrypoints-0.3-setup.py.patch (hunk 1 above)
Comment (a17r) [reply dev-python/entrypoints/files/entrypoints-0.3-setup.py.patch]: Please do it. It means that the automatically produced byproducts over at github are not meant as real release tarballs. Upstreams scripts are obviously adding stuff such as the missing setup.py.

File: dev-python/entrypoints/entrypoints-0.3.ebuild (hunk 2 above)
Comment (a17r) [reply dev-python/entrypoints/entrypoints-0.3.ebuild]: Maybe this won't be necessary anymore after having switched to the pypi release tarball.

**Comment by marbre** [comment]
@a17r Gentle ping.
